go/chat/server.go

```diff
@@ -762,17 +766,17 @@ func (h *Server) GetThreadNonblock(ctx context.Context, arg chat1.GetThreadNonbl
 			}
 			sJSONPt := string(jsonPt)
 			pthread = &sJSONPt
-			h.applyPagerModeOutgoing(bctx, arg.ConversationID, resThread.Pagination, pagination, arg.Pgmode)
+			h.applyPagerModeOutgoing(ctx, arg.ConversationID, resThread.Pagination, pagination, arg.Pgmode)
 		} else {
 			h.Debug(ctx, "GetThreadNonblock: sending nil cached response")
 		}
 		start := time.Now()
-		chatUI.ChatThreadCached(bctx, chat1.ChatThreadCachedArg{
+		chatUI.ChatThreadCached(ctx, chat1.ChatThreadCachedArg{
 			SessionID: arg.SessionID,
 			Thread:    pthread,
 		})
 		h.Debug(ctx, "GetThreadNonblock: cached response send time: %v", time.Since(start))
-	}()
+	}(localCtx)
 
 	wg.Add(1)
 	go func() {
```

Comment:
why doesn't this take a `localCtx` as a param?

Reply:
This is for the "remote" portion of the request, only the local portion is supposed to get canceled in this way.